<launch>

    <arg name="sim" default="true"/>
    <arg name="gui" default="true"/>

    <param name="/ardrone/image_raw/theora/quality" type="int" value="63"/>

    <include file="$(find ardrone_sim)/launch/ardrone_testworld.launch" if="$(arg sim)">
      <arg name="gui" value="$(arg gui)"/>
    </include>

    <!-- Launches the AR.Drone driver -->
    <node name="ardrone_driver" pkg="ardrone_autonomy" type="ardrone_driver" output="screen" clear_params="true" unless="$(arg sim)">
        <param name="outdoor" value="0" /> <!-- If we are flying outdoors, will select more aggressive default settings -->
        <param name="flight_without_shell" value="0" /> <!-- Changes internal controller gains if we are flying without the propeller guard -->

        <param name="altitude_max" value="3000" /> <!-- in millimeters = 3 meters = 9' -->
        <param name="altitude_min" value="50" /> <!-- in millimeters = 5cm = 2" -->
        <param name="euler_angle_max" value="0.1" /> <!-- maximum allowable body angle in radians = 5 degrees -->
        <param name="control_vz_max" value="200" /> <!-- maximum z velocity in mm/sec = 0.2m/sec -->
        <param name="control_yaw" value="1.0" /> <!-- maximum rotation rate in radians/sec = 40 degrees per second (1/9 rev/sec) -->
    </node>

</launch>
